pandora-client-web/src/components/gameContext/directoryConnectorContextProvider.tsx

```diff
@@ -143,3 +148,76 @@ export function useAuthTokenIsValid(): boolean {
 
 	return authToken != null && isValid;
 }
+
+function SecondFactorDialog() {
+	const directoryConnector = useDirectoryConnector();
+	const [types, setTypes] = React.useState<SecondFactorType[] | null>([]);
+	const [invalid, setInvalid] = React.useState<SecondFactorType[] | null>([]);
+	const [handler, setHandler] = React.useState<null | { resolve: (data: SecondFactorData | PromiseLike<SecondFactorData> | null) => void; }>(null);
```

Comment:
All three of these states are always set at the same time, it looks?
Wouldn't it then be much better and easier to combine them into a single state which is either `null` or object containing valid values of these three?